kafka-ui-api/src/main/java/com/provectus/kafka/ui/controller/TopicsController.java

```diff
@@ -72,35 +90,75 @@
   public Mono<ResponseEntity<TopicDetailsDTO>> getTopicDetails(
       String clusterName, String topicName, ServerWebExchange exchange) {
     return topicsService.getTopicDetails(getCluster(clusterName), topicName)
+        .map(clusterMapper::toTopicDetails)
         .map(ResponseEntity::ok);
   }
 
-  @Override
   public Mono<ResponseEntity<TopicsResponseDTO>> getTopics(String clusterName, @Valid Integer page,
                                                            @Valid Integer perPage,
                                                            @Valid Boolean showInternal,
                                                            @Valid String search,
                                                            @Valid TopicColumnsToSortDTO orderBy,
                                                            @Valid SortOrderDTO sortOrder,
                                                            ServerWebExchange exchange) {
-    return topicsService
-        .getTopics(
-            getCluster(clusterName),
-            Optional.ofNullable(page),
-            Optional.ofNullable(perPage),
-            Optional.ofNullable(showInternal),
-            Optional.ofNullable(search),
-            Optional.ofNullable(orderBy),
-            Optional.ofNullable(sortOrder)
-        ).map(ResponseEntity::ok);
+    return topicsService.getTopicsForPagination(getCluster(clusterName))
+        .flatMap(paginatingTopics -> {
```

Comment:
I would rename vars in this method : `paginatingTopics` -> `existingTopics`, `topics` -> `filtered`,  `filteredTopics` -> `topicsPage`